integrations/java/openhouse-java-runtime/src/main/java/com/linkedin/openhouse/javaclient/OpenHouseCatalog.java

```diff
@@ -128,6 +113,37 @@ public void initialize(String name, Map<String, String> properties) {
     this.cluster = properties.getOrDefault(CLUSTER_PROPERTY, DEFAULT_CLUSTER);
   }
 
+  /**
+   * sets the auth token as default header which gets added to every request from ApiClient
+   * @param token
+   */
+  protected void setToken(String token) {
+    if (token.isEmpty()){
+      this.properties.put(AUTH_TOKEN, token);
+      this.apiClient.addDefaultHeader("Authorization", String.format("Bearer %s", token));
+    }
+  }
+
+  private ApiClient createApiClient() {
```

Comment:
can we pass properties to this method from the initialize method? This method seems to be called only from the initialize method. Helps with code readability and clear relationship with the caller.

Reply:
Will wait for Sumedh's comment about keeping this refactoring out.

Reply:
Let's not mix fixing the code readability and adding token refresh functionality in the same PR. Fine with first doing the code readability PR before the functionality PR, your call.

Reply:
Removed the refactoring, and limited changes to tokenRefetch only.